engine/table/src/main/java/io/deephaven/engine/table/impl/TableDefaults.java

```diff
@@ -108,8 +108,13 @@ default <T> ColumnSource<T> getColumnSource(String sourceName, Class<? extends T
             @Nullable Class<?> componentType) {
         @SuppressWarnings("rawtypes")
         ColumnSource rawColumnSource = getColumnSource(sourceName);
-        // noinspection unchecked
-        return rawColumnSource.cast(clazz, componentType);
+        try {
+            // noinspection unchecked
+            return rawColumnSource.cast(clazz, componentType);
+        } catch (ClassCastException ex) {
+            throw new RuntimeException(
+                    "Error retrieving ColumnSource with type " + clazz.getName() + " for column " + sourceName, ex);
+        }
```

Comment:
That’s not better, or precise enough.
1. No throwing RTEs. Throw something more specific.
2. You should make it clearer that you’re throwing an exception because of a type mismatch. You would have failed before for a missing column.
3. I think the existing error messaging is clearer and more specific.

Reply:
A better change would just be to add an optional "name" prefix to `ColumnSource.cast`, so the existing exceptions could refer to "ColumnSource <name>".